### ISSUE TITLE
Resource 'aws_spot_instance_request' does not have attribute 'public_dns'

### ISSUE THREAD

**Issue by stolinzeev** [hashicorp/terraform#3960]
Having a trouble like https://github.com/hashicorp/terraform/issues/3154 but on terraform v0.6.6 and with EC2 (not VPC). 

That's my code:

```hcl
provider "aws" {
  access_key = "${var.access_key}"
  secret_key = "${var.secret_key}"
  region = "eu-west-1"
}

resource "aws_spot_instance_request" "server" {
  ami = "xxxxxxxx"
  instance_type = "m3.large"
  spot_price = "0.3"
  wait_for_fulfillment = true
  key_name = "xxxxxxxxx"
  iam_instance_profile = "xxxxxxx"
  security_groups = [
    "xxxxxx"
  ]
  user_data = "${file("xxxxxxxxx.txt")}"

  tags {
    Name="xxxxxxxx"
  }
}
resource "aws_route53_record" "server" {
  zone_id = "Z3B9RV5H90BA7G"
  name = "xxxxxxxxxx"
  type = "CNAME"
  ttl = "60"
  records = ["${aws_spot_instance_request.server.public_dns}"]
}
```

And Terraform responce:

```
aws_spot_instance_request.server: Creating...
  ami:                        "" => "ami-1390ac64"
  availability_zone:          "" => "<computed>"
  ebs_block_device.#:         "" => "<computed>"
  ephemeral_block_device.#:   "" => "<computed>"
  iam_instance_profile:       "" => "xxxxxx"
  instance_type:              "" => "m3.large"
  key_name:                   "" => "xxxxxx"
  placement_group:            "" => "<computed>"
  private_dns:                "" => "<computed>"
  private_ip:                 "" => "<computed>"
  public_dns:                 "" => "<computed>"
  public_ip:                  "" => "<computed>"
  root_block_device.#:        "" => "<computed>"
  security_groups.#:          "" => "1"
  security_groups.1744640391: "" => "xxxx"
  source_dest_check:          "" => "1"
  spot_bid_status:            "" => "<computed>"
  spot_instance_id:           "" => "<computed>"
  spot_price:                 "" => "0.3"
  spot_request_state:         "" => "<computed>"
  spot_type:                  "" => "persistent"
  subnet_id:                  "" => "<computed>"
  tags.#:                     "" => "1"
  tags.Name:                  "" => "xxxxxxxx"
  tenancy:                    "" => "<computed>"
  user_data:                  "" => "da39a3ee5e6b4b0d3255bfef95601890afd80709"
  vpc_security_group_ids.#:   "" => "<computed>"
  wait_for_fulfillment:       "" => "1"
aws_spot_instance_request.server: Creation complete
Error applying plan:

1 error(s) occurred:

* Resource 'aws_spot_instance_request.server' does not have attribute 'public_dns' for variable 'aws_spot_instance_request.server.public_dns'

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

**Comment by stack72** [comment]
@stolinzeev can you try the following code:

```
resource "aws_spot_instance_request" "server" {
  ami = "xxxxxxxx"
  instance_type = "m3.large"
  spot_price = "0.3"
  wait_for_fulfillment = true
  key_name = "xxxxxxxxx"
  iam_instance_profile = "xxxxxxx"
  associate_public_ip_address = true
  security_groups = [
    "xxxxxx"
  ]
  user_data = "${file("xxxxxxxxx.txt")}"

  tags {
    Name="xxxxxxxx"
  }
}
```

The addition of `associate_public_ip_address` works for me

**Comment by stack72** [comment]
@stolinzeev any joy with this?

**Comment by catsby** [comment]
Hello – I added `public_dns` to spot instances just a day or so before this was opened, I think, but it wasn't merged until the 23rd of November. `public_dns` is available in the latest release, v0.6.8

Thanks!
